plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/SnapshotsTable.java

```diff
@@ -81,29 +87,39 @@ private static List<Page> buildPages(ConnectorTableMetadata tableMetadata, Conne
     {
         PageListBuilder pagesBuilder = PageListBuilder.forTable(tableMetadata);
 
+        TableScan tableScan = buildTableScan(icebergTable, SNAPSHOTS);
         TimeZoneKey timeZoneKey = session.getTimeZoneKey();
-        icebergTable.snapshots().forEach(snapshot -> {
-            pagesBuilder.beginRow();
-            pagesBuilder.appendTimestampTzMillis(snapshot.timestampMillis(), timeZoneKey);
-            pagesBuilder.appendBigint(snapshot.snapshotId());
-            if (checkNonNull(snapshot.parentId(), pagesBuilder)) {
-                pagesBuilder.appendBigint(snapshot.parentId());
-            }
-            if (checkNonNull(snapshot.operation(), pagesBuilder)) {
-                pagesBuilder.appendVarchar(snapshot.operation());
-            }
-            if (checkNonNull(snapshot.manifestListLocation(), pagesBuilder)) {
-                pagesBuilder.appendVarchar(snapshot.manifestListLocation());
-            }
-            if (checkNonNull(snapshot.summary(), pagesBuilder)) {
-                pagesBuilder.appendVarcharVarcharMap(snapshot.summary());
-            }
-            pagesBuilder.endRow();
+
+        tableScan.planFiles().forEach(fileScanTask -> {
+            DataTask dataTask = (DataTask) fileScanTask;
+            dataTask.rows().forEach(structLike -> addRow(pagesBuilder, structLike, timeZoneKey));
         });
 
         return pagesBuilder.build();
     }
 
+    private static void addRow(PageListBuilder pagesBuilder, StructLike structLike, TimeZoneKey timeZoneKey)
+    {
+        pagesBuilder.beginRow();
+        pagesBuilder.appendTimestampTzMillis((castToMillis(structLike.get(0, Long.class))), timeZoneKey);
```

Comment:
I'd just inline this, use the 1000 constant, it's a little more readable
```suggestion
        pagesBuilder.appendTimestampTzMillis(structLike.get(0, Long.class) / MICROSECONDS_PER_MILLISECOND, timeZoneKey);
```